### PR TITLE
NTR - Reservoir Host

### PR DESCRIPTION
**Pull request by jhpoelen**
Hi!

@wdduncan et al. I finally got around to add the proposed/discussed reservoir host of RO:0002802 / has reservoir host RO:0002803 terms. See https://github.com/oborel/obo-relations/issues/368 . 

Thanks for being patient in guiding me through the term proposal process.

My edits use previously unused ids in my own id range (i.e., RO:0002802 / RO:0002803 ). 

My edits appear to have passed the CI workflow on my fork prior to creating this pull request. For CI report, see https://github.com/jhpoelen/obo-relations/actions/runs/6015836903 and/or attached screenshot below.

I've added myself and @kephelps as term editor. @kephelps is subject matter expert, and I am more of the technical side of things. @kephelps helped find a suitable definition of the term in existing literature, I helped integrate this into ro via attached edits.

Please confirm receipt of this pull request and let me know if there's any more I need to do to complete the NTR process.

Hope this helps,
-jorrit

image

### CODE REVIEW

**Comment by jhpoelen** [comment]
hi @anitacaron - thanks for taking action on this pull request. Is there anything I should/can do to help move this along?

**Comment by jhpoelen** [comment]
Perhaps @cmungall can help review? Or @balhoff ? Open to suggestions. . .

File: src/ontology/ro-edit.owl [hunk 1]
```diff
@@ -5329,6 +5331,25 @@ AnnotationAssertion(rdfs:label obo:RO_0002801 "co-roosts with"@en)
...
+AnnotationAssertion(obo:IAO_0000117 obo:RO_0002802 <https://orcid.org/0000-0003-3138-4118>)
+AnnotationAssertion(obo:IAO_0000117 obo:RO_0002802 <https://orcid.org/0000-0002-3120-4802>)
+AnnotationAssertion(obo:IAO_0000119 obo:RO_0002802 <https://doi.org/10.3390/v13071356>)
+AnnotationAssertion(rdfs:label obo:RO_0002802 "reservoir host of")
+SubObjectPropertyOf(obo:RO_0002802 obo:RO_0002453)
+InverseObjectProperties(obo:RO_0002802 obo:RO_0002803)
+
+# Object Property: obo:RO_0002803 (has reservoir host)
+
+AnnotationAssertion(obo:IAO_0000115 obo:RO_0002803 "inverse of reservoir host of")
```
[the remark this answers is not in the document]
Comment (jhpoelen) [reply src/ontology/ro-edit.owl:5346]: @anitacaron thanks for taking the time to review. Can you please elaborate on your question - 

> Is the definition here should be in the other term RO:0002802?

I am not sure I understand your question.

File: src/ontology/ro-edit.owl [hunk 2]
```diff
@@ -5329,6 +5331,25 @@ AnnotationAssertion(rdfs:label obo:RO_0002801 "co-roosts with"@en)
...
+AnnotationAssertion(obo:IAO_0000117 obo:RO_0002802 <https://orcid.org/0000-0003-3138-4118>)
+AnnotationAssertion(obo:IAO_0000117 obo:RO_0002802 <https://orcid.org/0000-0002-3120-4802>)
+AnnotationAssertion(oboInOwl:inSubset obo:RO_0002803 obo:RO_0002259)
+AnnotationAssertion(rdfs:label obo:RO_0002803 "has reservoir host")
+SubObjectPropertyOf(obo:RO_0002803 obo:RO_0002454)
```
[the remark this answers is not in the document]
Comment (jhpoelen) [reply src/ontology/ro-edit.owl:5353]: I've added an example via https://github.com/jhpoelen/obo-relations/commit/018eafb3de142dc1b7e55d76f72d0c3dab4d7f13 .

File: src/ontology/ro-edit.owl [hunk 3]
```diff
@@ -5329,6 +5331,25 @@ AnnotationAssertion(rdfs:label obo:RO_0002801 "co-roosts with"@en)
...
+AnnotationAssertion(obo:IAO_0000119 obo:RO_0002802 <https://doi.org/10.3390/v13071356>)
+AnnotationAssertion(rdfs:label obo:RO_0002802 "reservoir host of")
+SubObjectPropertyOf(obo:RO_0002802 obo:RO_0002453)
+InverseObjectProperties(obo:RO_0002802 obo:RO_0002803)
+
+# Object Property: obo:RO_0002803 (has reservoir host)
+
+AnnotationAssertion(obo:IAO_0000115 obo:RO_0002803 "inverse of reservoir host of")
+AnnotationAssertion(obo:IAO_0000117 obo:RO_0002802 <https://orcid.org/0000-0003-3138-4118>)
+AnnotationAssertion(obo:IAO_0000117 obo:RO_0002802 <https://orcid.org/0000-0002-3120-4802>)
```
[the remark this answers is not in the document]
Comment (jhpoelen) [reply src/ontology/ro-edit.owl]: applied as suggested via https://github.com/jhpoelen/obo-relations/commit/a7f44c749a2ff00877dde7696d08cf8d2ea57543

File: src/ontology/ro-edit.owl [hunk 4]
```diff
@@ -5329,6 +5331,25 @@ AnnotationAssertion(rdfs:label obo:RO_0002801 "co-roosts with"@en)
 SubObjectPropertyOf(obo:RO_0002801 obo:RO_0008506)
 SymmetricObjectProperty(obo:RO_0002801)
 
+# Object Property: obo:RO_0002802 (reservoir host of)
+
+AnnotationAssertion(obo:IAO_0000115 obo:RO_0002802 "A reservoir is a population, species or community (assemblage of different species in a given geographic area) in which a microorganism naturally occurs and is indefinitely maintained. Some zoonotic pathogens, particularly bacterial pathogens, may also have environmental reservoirs. Microorganisms with multiple reservoir species may be indefinitely maintained across the community of species even if they are not always present in each individual reservoir species. In the reservoir species or community, the microorganism may cause either asymptomatic infection or disease and this may vary among individuals. A pathogen may also be more genetically diverse in its reservoir host than in other taxa, in part because the pathogen is endemic to reservoir hosts.")
+AnnotationAssertion(obo:IAO_0000117 obo:RO_0002802 <https://orcid.org/0000-0003-3138-4118>)
+AnnotationAssertion(obo:IAO_0000117 obo:RO_0002802 <https://orcid.org/0000-0002-3120-4802>)
```
[the remark this answers is not in the document]
Comment (jhpoelen) [reply src/ontology/ro-edit.owl]: applied as suggested via https://github.com/jhpoelen/obo-relations/commit/a7f44c749a2ff00877dde7696d08cf8d2ea57543

File: src/ontology/ro-edit.owl (hunk 3 above)
[the remark this answers is not in the document]
Comment (jhpoelen) [reply src/ontology/ro-edit.owl]: @anitacaron please mark as resolved is this change works for you

File: src/ontology/ro-edit.owl (hunk 2 above)
[the remark this answers is not in the document]
Comment (jhpoelen) [reply src/ontology/ro-edit.owl:5353]: @anitacaron please mark as resolved is this change works for you

File: src/ontology/ro-edit.owl (hunk 4 above)
[the remark this answers is not in the document]
Comment (jhpoelen) [reply src/ontology/ro-edit.owl]: @anitacaron please mark as resolved is this change works for you

File: src/ontology/ro-edit.owl (hunk 2 above)
[the remark this answers is not in the document]
Comment (anitacaron) [reply src/ontology/ro-edit.owl:5353]: Thank you, @jhpoelen.

File: src/ontology/ro-edit.owl [hunk 5]
```diff
@@ -5329,6 +5331,27 @@ AnnotationAssertion(rdfs:label obo:RO_0002801 "co-roosts with"@en)
...
+AnnotationAssertion(obo:IAO_0000112 obo:RO_0002802 "An individual of species Camelus dromedarius (commonly known as dromedary camels) is a reservoir host of an individual of Middle East respiratory syndrome–related coronavirus (MERS-CoV) since at least the 1980s.")
+AnnotationAssertion(terms:contributor obo:RO_0002802 <https://orcid.org/0000-0003-3138-4118>)
+AnnotationAssertion(terms:contributor obo:RO_0002802 <https://orcid.org/0000-0002-3120-4802>)
+AnnotationAssertion(obo:IAO_0000119 obo:RO_0002802 <https://doi.org/10.3390/v13071356>)
+AnnotationAssertion(rdfs:label obo:RO_0002802 "reservoir host of")
+SubObjectPropertyOf(obo:RO_0002802 obo:RO_0002453)
+InverseObjectProperties(obo:RO_0002802 obo:RO_0002803)
```
Comment (anitacaron) [inline src/ontology/ro-edit.owl:5343]: This logical axiom should be on the property that says it's inverse.

File: src/ontology/ro-edit.owl (hunk 4 above)
[the remark this answers is not in the document]
Comment (anitacaron) [reply src/ontology/ro-edit.owl]: Thank, @jhpoelen.

File: src/ontology/ro-edit.owl (hunk 3 above)
[the remark this answers is not in the document]
Comment (anitacaron) [reply src/ontology/ro-edit.owl]: Thank you, @jhpoelen.

File: src/ontology/ro-edit.owl (hunk 1 above)
[the remark this answers is not in the document]
Comment (anitacaron) [reply src/ontology/ro-edit.owl:5347]: This property sounds more direct than the inverse form of `reservoir host of`.

File: src/ontology/ro-edit.owl (hunk 5 above)
Comment (jhpoelen) [reply src/ontology/ro-edit.owl:5343]: @anitacaron Thanks again for taking the time to comment and for humoring my relative naivete . 

with your comment:
> This logical axiom should be on the property that says it's inverse.

do you mean that I should move the axiom to the obo:RO_0002803? 

I'd appreciate suggestions, I am still catching up with the ontology jargon.

File: src/ontology/ro-edit.owl (hunk 1 above)
[the remark this answers is not in the document]
Comment (jhpoelen) [reply src/ontology/ro-edit.owl:5347]: @anitacaron thanks for elaborating. I am still a bit puzzled on what you are suggesting. 

How can I add an annotation to document that the property is more direct than the inverse form of `reservoir host of` ? Can you give an example?

Thanks for being patient with me?

**Comment by jhpoelen** [comment]
Just to check in - I have made the changes I understand how to make, and realize that I may need to to address remaining comments that @anitacaron shared. E.g., https://github.com/oborel/obo-relations/pull/757#discussion_r1312956720 . 

Is anyone willing and able to help me figure this out with the goal to get the term [reservoir host] admitted to the Relations Ontology? 

What are additional steps I can take to help move this forward?

**Comment by jhpoelen** [comment]
@wdduncan suggest to shorten the existing definition in genus-species differential form, using other RO terms as an example. Suggest to use rdfs:comment to move the current definition in.

File: src/ontology/ro-edit.owl (hunk 1 above)
[the remark this answers is not in the document]
Comment (anitacaron) [reply src/ontology/ro-edit.owl:5347]: After discussion on 2023-10-09 Relation Ontology Meeting, it's just necessary to move the axiom `InverseObjectProperties(obo:RO_0002803 obo:RO_0002802)` to the object property `has reservoir host`.

File: src/ontology/ro-edit.owl (hunk 5 above)
Comment (jhpoelen) [reply src/ontology/ro-edit.owl]: @anitacaron I've moved the logical axiom to be on the property that says it's inverse as you suggested. Please review https://github.com/oborel/obo-relations/pull/757/commits/a7ea89c83d4741f35d57c63d1e649a6779d2de0e and resolve our conversation or make suggestion for further improvements.

Thanks for reviewing and for being patient with me.

File: src/ontology/ro-edit.owl [hunk 6]
```diff
@@ -5333,8 +5333,9 @@ SymmetricObjectProperty(obo:RO_0002801)
 
 # Object Property: obo:RO_0002802 (reservoir host of)
 
-AnnotationAssertion(obo:IAO_0000115 obo:RO_0002802 "A reservoir is a population, species or community (assemblage of different species in a given geographic area) in which a microorganism naturally occurs and is indefinitely maintained. Some zoonotic pathogens, particularly bacterial pathogens, may also have environmental reservoirs. Microorganisms with multiple reservoir species may be indefinitely maintained across the community of species even if they are not always present in each individual reservoir species. In the reservoir species or community, the microorganism may cause either asymptomatic infection or disease and this may vary among individuals. A pathogen may also be more genetically diverse in its reservoir host than in other taxa, in part because the pathogen is endemic to reservoir hosts.")
+AnnotationAssertion(obo:IAO_0000115 obo:RO_0002802 "A sub-relation of host-of in which the host is a reservoir in which the hosted organism naturally occurs and is indefinitely maintained.")
```
Comment (jhpoelen) [inline src/ontology/ro-edit.owl:5336]: @wdduncan please review my edits following your suggestion to shorten the definition of reservoir host and move the quoted definition to rdfs:comment.

Comment (ddooley) [reply src/ontology/ro-edit.owl:5336]: Possibly "A host-of relation which attaches between a host organism and a hosted organism that naturally occurs in an indefinitely maintained reservoir that the host provides."

**Comment by wdduncan** [comment]
Hi @jhpoelen. Thanks for all the edits. There was a lot of activity since the RO call! 

I like the comment you added. Can you add a link or reference to the source of the information (e.g., [Wikipedia](https://en.wikipedia.org/wiki/Natural_reservoir)).

I like the way you defined 'reservoir' in the comment. However I'm a little apprehensive about your use of the word 'reservoir'. The word 'reservoir' has different meanings depending on context. Can you use 'reservoir species' or 'environmental reservoir' to make the context more clear? If your use case is primarily dealing with pathogens, then maybe 'disease reservoir' is the best option? E.g.:

```
A relation between a disease reservoir and an organism in which the organism naturally lives and reproduces, or primarily depends for its survival on the disease reservoir.
```

**Comment by jhpoelen** [comment]
@wdduncan Thanks for your comments:

you said - 
> Can you add a link or reference to the source of the information (e.g., [Wikipedia](https://en.wikipedia.org/wiki/Natural_reservoir)).

I've added an additional definition source, now including an expanded citation of the paper that contains the definition:

Shapiro, J.T.; Víquez-R, L.; Leopardi, S.; Vicente-Santos, A.; Mendenhall, I.H.; Frick, W.F.; Kading, R.C.; Medellín, R.A.; Racey, P.; Kingston, T. Setting the Terms for Zoonotic Diseases: Effective Communication for Research, Conservation, and Public Policy. Viruses 2021, 13, 1356. https://doi.org/10.3390/v13071356

you said

> I'm a little apprehensive about your use of the word 'reservoir'. The word 'reservoir' has different meanings depending on context. Can you use 'reservoir species' or 'environmental reservoir' to make the context more clear? If your use case is primarily dealing with pathogens, then maybe 'disease reservoir' is the best option? If so, I would change the label to 'disease reservoir host for'. [...]

I'll have to get back to you on that, I am consulting my peers.

File: src/ontology/ro-edit.owl (hunk 5 above)
Comment (anitacaron) [reply src/ontology/ro-edit.owl]: Thanks @jhpoelen!

File: src/ontology/ro-edit.owl [hunk 7]
```diff
@@ -5329,6 +5331,29 @@ AnnotationAssertion(rdfs:label obo:RO_0002801 "co-roosts with"@en)
...
+AnnotationAssertion(terms:contributor obo:RO_0002802 <https://orcid.org/0000-0003-3138-4118>)
+AnnotationAssertion(terms:contributor obo:RO_0002802 <https://orcid.org/0000-0002-3120-4802>)
+AnnotationAssertion(obo:IAO_0000119 obo:RO_0002802 <https://doi.org/10.3390/v13071356>)
+AnnotationAssertion(obo:IAO_0000119 obo:RO_0002802 "Shapiro, J.T.; Víquez-R, L.; Leopardi, S.; Vicente-Santos, A.; Mendenhall, I.H.; Frick, W.F.; Kading, R.C.; Medellín, R.A.; Racey, P.; Kingston, T. Setting the Terms for Zoonotic Diseases: Effective Communication for Research, Conservation, and Public Policy. Viruses 2021, 13, 1356. https://doi.org/10.3390/v13071356")
```
Comment (anitacaron) [inline src/ontology/ro-edit.owl:5342]: Please remove the expanded citation, only the doi link is enough.

**Review by anitacaron — CHANGES_REQUESTED** [review]

**Comment by jhpoelen** [comment]
@wdduncan in discussion with peers no clear consensus was reached - some argued to use "host of" instead, while others said that the term gains specific meaning from a disease context. 

So, with no clear consensus, I'd like to revert to the source definition from 

Shapiro, J.T.; Víquez-R, L.; Leopardi, S.; Vicente-Santos, A.; Mendenhall, I.H.; Frick, W.F.; Kading, R.C.; Medellín, R.A.; Racey, P.; Kingston, T. Setting the Terms for Zoonotic Diseases: Effective Communication for Research, Conservation, and Public Policy. Viruses 2021, 13, 1356. https://doi.org/10.3390/v13071356

as referenced via 

```
AnnotationAssertion(obo:IAO_0000119 obo:RO_0002802 <https://doi.org/10.3390/v13071356>)
```

This definition does *not* explicitly refer to the reservoir host as a disease reservoir host. Instead, the definition uses "micro organism" instead of "pathogen". 

This is why I'd like to advocated to keep the definition as is, and have the context be added through their use in combination with other relations. For instance, if a bat is a host reservoir for some corona virus, and that corona virus is a pathogen for cats, than, you can infer that a bat is a disease reservoir for a cat pathogen. 

@wdduncan curious to hear your thoughts.

**Comment by wdduncan** [comment]
Hi @jhpoelen 

> This definition does not explicitly refer to the reservoir host as a disease reservoir host. Instead, the definition uses "micro organism" instead of "pathogen".

Ok. I wasn't sure if your use case was for disease causing organisms or could be more general. I would use 'organism' instead of 'microorganism', to stay clear of debates about what makes an organism 'micro'.

Based on this definition:
> A host-of relation which attaches between a host organism and a hosted organism where the hosted organism naturally occurs in an indefinitely maintained reservoir that the host provides.

If so, I would reword slightly to:
> "A relation between a host organism and a hosted organism in which the hosted organism naturally occurs in an indefinitely maintained reservoir provided by the host.

In my amateur mind, I understand 'maintained reservoir' to be some kind of naturally occurring environment within the host that is suitable for the hosted organism. If so, you may want to say '...  an indefinitely maintained **environmental** reservoir provided by the host'. (You may need to adjust your comment too.)

**Comment by jhpoelen** [comment]
I've committed @wdduncan reworded definition via https://github.com/oborel/obo-relations/pull/757/commits/3e84ddde09517b4e9b1ead1ffa0630eef5cff783 .

@anitacaron @wdduncan @ddooley Please confirm that this resolves the last remaining concern by reviewers by accepting my  pull request.

**Review by wdduncan — APPROVED** [review]

**Review by ddooley — APPROVED** [review]

**Review by anitacaron — APPROVED** [review]

**Comment by jhpoelen** [comment]
Thanks to @anitacaron @wdduncan and @ddooley for taking the time to review and offer suggestions to improve the term definitions and annotations.

**Comment by jhpoelen** [comment]
Thanks @anitacaron for creating the RO release https://github.com/oborel/obo-relations/releases/tag/v2024-02-13 with the reservoir host term in it! 

Yay!
